src/scenario_tests.rs

```diff
@@ -1270,6 +1271,54 @@ mod src_tests {
         );
     }
 
+    #[test]
+    fn parsec_check_get_node_to_relocate_uses_online_nodes_only() {
+        let mut all_non_online_states = vec![
+            State::RelocatingAgeIncrease,
+            State::RelocatingHop,
+            State::RelocatingBackOnline,
+            State::Relocated(get_relocated_info(CANDIDATE_1_OLD, SECTION_INFO_1)),
+            State::WaitingCandidateInfo(get_relocated_info(CANDIDATE_2_OLD, SECTION_INFO_2)),
+            State::WaitingProofing,
+            State::Offline,
+        ];
+
+        let age = CANDIDATE_1_OLD.0.age;
+        let states_with_enough_work_units = all_non_online_states
+            .drain(..)
+            .enumerate()
+            .map(|(name_increment, state)| {
+                let node = Node(Attributes {
```

Comment:
I'd probably put that directly in NodeState as in.

```
NodeState {
                    node : Node(Attributes {
                        name: 1000 + name_increment as i32,
                        age,
                    },
                    is_elder: false,
                    work_units_done: age,
                    state,
                }
```

No strong opinion though.